util/offsetfix.C

```diff
@@ -465,14 +466,11 @@ void GetTimeDiff(std::vector<TEventTime*> *eventQ, int64_t *correction){
       if(mapit->first == TEventTime::GetBestDigitizer())
          continue;
       fillhist = (TH1D*)(list->At(mapit->second));
-      std::cout << static_cast<int64_t>(fillhist->GetBinCenter(fillhist->GetMaximumBin())) << std::endl;
-      correction[mapit->second] +=  static_cast<int64_t>(fillhist->GetBinCenter(fillhist->GetMaximumBin()));
-      TPolyMarker *pm = new TPolyMarker;
-      pm->SetNextPoint(fillhist->GetBinCenter(fillhist->GetMaximumBin()),fillhist->GetBinContent(fillhist->GetMaximumBin())+10);
-      pm->SetMarkerStyle(23);
-      pm->SetMarkerColor(kRed);
-      pm->SetMarkerSize(1.3);
-      fillhist->GetListOfFunctions()->Add(pm);
+      TSpectrum* spec = new TSpectrum();
+      spec->Search(fillhist);
+      double peak = spec->GetPositionX()[0];
+      std::cout << static_cast<int64_t>(peak) << std::endl;
+      correction[mapit->second] +=  static_cast<int64_t>(peak);
```

Comment:
I'm not sure how the search returns the double, but the "correct" way to convert a double to an int is 
```c++
static_cast<int64_t>floor(peak+0.5);
```
That way it adds 0.5 and then truncates. This rounds to the appropriate int. Otherwise c++ will just straight up truncate. and we could be wrong by a whole bin. I think it can also sometimes convert something like 3.0000 to 2 just because of the way the double's bits are set.